### PR TITLE
Add Queue Reference

### PR DESCRIPTION
**Pull request by artfulrobot**
I found it hard to figure out how queues work as it was not included in the docs. Here is a single page reference which includes

- basic concepts
- links to the demoqueue extension (which I wish I'd found sooner!)
- cautions for programmers (based on painful experience!)

### CODE REVIEW

File: docs/framework/queues/index.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
+# Queues Reference
+
+CiviCRM has a system for splitting big jobs into smaller tasks on a queue, and
```
Comment (MikeyMJCO) [inline docs/framework/queues/index.md]:
```suggestion
CiviCRM has a system for splitting big jobs into smaller tasks on a queue, and provides a couple of ways to process queues.
```

Leave line wrapping to the editors/ui :-)

File: docs/framework/queues/index.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+implemented(!). You can have `Memory` or `Sql`.
+
+Example:
+
+```php
+$queue = CRM_Queue_Service::singleton()->create([
+      'type'  => 'Sql',
+      'name'  => 'my-own-queue',
+    ]);
+
```
Comment (MikeyMJCO) [inline docs/framework/queues/index.md]:
```suggestion
```

**Comment by MikeyMJCO** [comment]
Merging!

**Comment by totten** [comment]
Good idea adding a chapter to the dev-docs, @artfulrobot !

**Comment by seamuslee001** [comment]
@artfulrobot @MikeyMJCO @totten i just happened to be playing with queues but there is actually one implementation that was missed which is what @jaapjansma actually did ages ago which is if you look at https://wiki.civicrm.org/confluence/display/CRMDOC/Howto+use+the+Queue+mechanism+in+your+extension and look at Step 7 you can see you can run a limited number of queue jobs through a cron job type mechanism

**Comment by seamuslee001** [comment]
@artfulrobot it seems Civirules uses it for their 'delayed' actions and added content to #675

**Pull request merged**